package/network/config/netifd/files/lib/netifd/dhcp.script

```diff
@@ -20,7 +20,13 @@ setup_interface () {
 
 	local i
 	for i in $router; do
-		proto_add_ipv4_route "$i" 32 "" "$ip"
+		local ip_net
+		local gw_net
+		ip_net=$(ipcalc.sh $ip/$mask | grep "NETWORK" | awk -F '=' '{print $2}')
```

Comment:
Usage of grep and awk is awkward; use eval "$(ipcalc.sh $ip/$mask)"; ip_net="$NETWORK".
Bring the ip_net calculation out of the loop as it does not make sense to calculate it for every loop iteration